changes/2020-07-14_multi-keyring-require-generation/change.md

```diff
@@ -0,0 +1,108 @@
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The multi-keyring has a clearly configured intent of which keyring SHOULD be used to generate the data key,
+but this is not actually enforced during the encryption operation.
+This change strengthens the requirements on multi-keyrings such that if a generator is configured,
+it MUST generate the data key.
+
+## Out of Scope
+
+Detecting at keyring configuration time that encryption will always fail
+because keyrings are not capable of meeting these requirements
+is [tracked separately](https://github.com/awslabs/aws-encryption-sdk-specification/issues/144).
+
+## Motivation
+
+If we do not make a guarantee that the generator will generate,
+then its usefulness is limited.
+Customers may have a compliance requirement that all data keys MUST be generated by a specific HSM,
+or may want to lock down true least-privilege permissions for their AWS KMS key policies.
+Under the current model,
+customers have to be very careful to make sure that they know
+exactly where and how a given multi-keyring configuration is being used
+rather than being certain that it will always do the same thing no matter what.
+
+## Drawbacks
+
+This is a breaking change,
+since existing multi-keyring configurations
+can be used in multiple contexts,
+and data key generation is not tightly controlled.
+We judge this to be the right trade-off
+between security and usability.
+
+## Security Implications
+
+Removing this flexibility may help identify potential security issues,
+as customers may discover when upgrading their ESDK version
+that data keys are not being generated as intended.
+Correcting this will allow customers
+to improve their security postures
+through improvements such as scoping down permissions.
+
+## Operational Implications
+
+Customers that are currently using a single multi-keyring instance
+in multiple contexts with inconsistent data key generation patterns
+will encounter errors when upgrading.
+This may lead to an increase in support requests,
+but should also reduce troubleshooting requests
+related to unexpected behaviour around data key generation,
+or unexpected permissions errors due to the same.
+
+## Guide-level Explanation
+
+The description of OnEncrypt for the multi-keyring
+will be changed to read as follows:
+
+If this keyring has a generator keyring,
+this keyring MUST first generate a plaintext data key using the generator keyring.
+If the input encryption materials already include a plaintext data key,
+OnEncrypt MUST fail.
+Otherwise,
+this keyring MUST first call that generator keyring's OnEncrypt
+using the input encryption materials as input.
+If the generator keyring fails OnEncrypt,
+this OnEncrypt MUST also fail.
```

Comment:
```suggestion
If this keyring has a generator keyring,
and the input encryption materials includes a plaintext data key,
OnEncrypt MUST fail.
This keyring MUST first call that generator keyring's OnEncrypt
using the input encryption materials as input.
If the generator keyring fails OnEncrypt,
this OnEncrypt MUST also fail.
```

Reply:
Ended up using bullet points to clarify the scoping here instead